Test suite failure - TestCorrelations

See for example: https://github.com/hyperspy/hyperspy-extensions-list/actions/runs/2587111143

```python
=================================== FAILURES ===================================
_________________ TestCorrelations.test_correlations_wrapping __________________
self = <pyxem.tests.utils.test_correlation_utils.TestCorrelations object at 0x7f61439a7610>
ones_hundred = array([[[100](https://github.com/hyperspy/hyperspy-extensions-list/runs/7123793736?check_suite_focus=true#step:16:101)., 100., 100., 100., 100., 100., 100., 100., 100., 100., 100.,
        100., 100., 100., 100., 100., 100., ...,   1.,   1.,   1.,   1.,   1.,   1.,   1.,   1.,   1.,
          1.,   1.,   1.,   1.,   1.,   1.,   1.,   1.,   1.]])
    def test_correlations_wrapping(
        self, ones_hundred
    ):  # Need to do extra checks to assure this is correct
        m = np.zeros((10, 20))
        m[2:4, :] = 1
>       c = _correlation(ones_hundred, axis=0, normalize=True, wrap=False)
/usr/share/miniconda3/lib/python3.9/site-packages/pyxem/tests/utils/test_correlation_utils.py:81: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
z = array([[  0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,
          0.,   0.,   0.,   0.,   0.,   0., ...,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,
          0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.]])
axis = 0
mask = array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0.],
       [0., 0., 0., 0...     0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0.]])
wrap = False, normalize = True
    def _correlation(z, axis=0, mask=None, wrap=True, normalize=True):
        r"""A generic function for applying a correlation with a mask.
        Takes a nd image and then preforms a auto-correlation on some axis.
        Used in the electron correlation and angular correlation codes. Uses
        the fft to speed up the correlation.
        Parameters
        ----------
        z: np.array
            A nd numpy array
        axis: int
            The axis to apply the correlation to
        mask: np.array
            A boolean array of the same size as z
        wrap: bool
            Allow the function to wrap or add zeros to the beginning and the
            end of z on the specified axis
        normalize: bool
            Subtract <I(\theta)>^2 and divide by <I(\theta)>^2
        """
        if wrap is False:
            z_shape = np.shape(z)
            padder = [(0, 0)] * len(z_shape)
            pad = z_shape[axis]  # This will be faster if the length of the axis
            # is a power of 2.  Based on the numpy implementation.  Not terribly
            # faster I think..
            padder[axis] = (pad, pad)
            slicer = [
                slice(None),
            ] * len(z_shape)
            slicer[axis] = slice(0, -2 * pad)  # creating the proper slices
            if mask is None:
                mask = np.zeros(shape=np.shape(z))
            z = np.pad(z, padder, "constant")
        if mask is not None or wrap is False:  # Need to scale for wrapping properly
            m = np.array(mask, dtype=bool)  # make sure it is a boolean array
            # this is to determine how many of the variables were non zero... This is really dumb.  but...
            # it works and I should stop trying to fix it (wreak it)
            mask_boolean = ~m  # inverting the boolean mask
            if wrap is False:  # padding with zeros to the function along some axis
                m = np.pad(
                    m, padder, "constant"
                )  # all the zeros are masked (should account for padding
                #  when normalized.
                mask_boolean = np.pad(mask_boolean, padder, "constant")
            mask_fft = np.fft.rfft(mask_boolean, axis=axis)
            number_unmasked = np.fft.irfft(
                mask_fft * np.conjugate(mask_fft), axis=axis
            ).real
            number_unmasked[
                number_unmasked < 1
            ] = 1  # get rid of divide by zero error for completely masked rows
            z[m] = 0
        # fast method uses a FFT and is a process which is O(n) = n log(n)
        I_fft = np.fft.rfft(z, axis=axis)
        a = np.fft.irfft(I_fft * np.conjugate(I_fft), axis=axis)
        if mask is not None:
            a = np.multiply(np.divide(a, number_unmasked), np.shape(z)[0])
        if normalize:  # simplified way to calculate the normalization
            row_mean = np.mean(a, axis=axis)
            row_mean[row_mean == 0] = 1
            row_mean = np.expand_dims(row_mean, axis=axis)
            a = np.divide(np.subtract(a, row_mean), row_mean)
        if wrap is False:
            print(slicer)
>           a = a[slicer]
E           IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
/usr/share/miniconda3/lib/python3.9/site-packages/pyxem/utils/correlation_utils.py:75: IndexError
```

--- a/pyxem/utils/correlation_utils.py
+++ b/pyxem/utils/correlation_utils.py
@@ -31,7 +31,7 @@ def _correlation(z, axis=0, mask=None, wrap=True, normalize=True):
         padder[axis] = (pad, pad)
         slicer = [
             slice(None),
-        ] * len(z_shape)
+                 ] * len(z_shape)
         slicer[axis] = slice(0, -2 * pad)  # creating the proper slices
         if mask is None:
             mask = np.zeros(shape=np.shape(z))
@@ -71,8 +71,7 @@ def _correlation(z, axis=0, mask=None, wrap=True, normalize=True):
         a = np.divide(np.subtract(a, row_mean), row_mean)
 
     if wrap is False:
-        print(slicer)
-        a = a[slicer]
+        a = a[tuple(slicer)]
     return a
 
 

